go/vt/vttablet/tabletmanager/vreplication/replicator_plan.go

```diff
@@ -444,6 +446,120 @@ func (tp *TablePlan) applyChange(rowChange *binlogdatapb.RowChange, executor fun
 	return nil, nil
 }
 
+// applyBulkDeleteChanges applies a bulk delete statement from the row changes
+// to the target table using an IN clause with the primary key values of the
+// rows to be deleted. This only supports tables with single column primary keys.
+func (tp *TablePlan) applyBulkDeleteChanges(rowDeletes []*binlogdatapb.RowChange, executor func(string) (*sqltypes.Result, error), maxQuerySize int64) (*sqltypes.Result, error) {
+	if len(rowDeletes) == 0 {
+		return nil, nil
+	}
+	if (len(tp.TablePlanBuilder.pkCols) + len(tp.TablePlanBuilder.extraSourcePkCols)) != 1 {
+		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "bulk delete is only supported for tables with a single primary key column")
```

Comment:
We don't want to extend this here for composite primary keys? Or is that for a separate PR?

Reply:
Added a TODO for now. That would require much more testing as the performance of those DELETEs can be unexpected based on past experience. That concern may not be warranted anymore with 8.0+, but for a related recent example with MariaDB:
https://jira.mariadb.org/browse/MDEV-32878